### PR TITLE
feat: added UserController

### PR DESCRIPTION
**Pull request by DFraczek**
### 1. **UserController Implementation**
   - Added a new `UserController` to manage user-related operations. This includes endpoints for:
     - **Updating user roles** (`PUT /api/user/roles/{username}`) – Only accessible by ADMIN users.
     - **Fetching user information by username** (`GET /api/user/username/{username}`).
     - **Fetching user information by email** (`GET /api/user/email/{email}`).
     - **Fetching user information by user ID** (`GET /api/user/id/{id}`).
     - **Deleting a user by ID** (`DELETE /api/user/{id}`) – Only accessible by ADMIN users.

### 2. **Book and Movie Unit Tests Fixes**
   - **BookNotFoundException** and **MovieNotFoundException** have been introduced to replace `EntityNotFoundException` for Book and Movie deletion and retrieval operations.

## Changes Summary:
- Created **UserController** to handle user role management and user details retrieval.
- Fixed unit tests to properly handle `BookNotFoundException` and `MovieNotFoundException`.

### CODE REVIEW

File: library/src/main/java/org/pollub/library/user/controller/UserController.java [hunk 1]
```diff
@@ -24,4 +25,32 @@ public ResponseEntity<ApiTextResponse> updateRoles(@PathVariable String username
...
+
+    @GetMapping("/email/{email}")
+    @PreAuthorize("isAuthenticated()")
+    public ResponseEntity<User> getUserByEmail(@PathVariable String email) {
+        User user = userService.findByEmail(email);
+        return ResponseEntity.ok(user);
+    }
+
+    @GetMapping("/id/{id}")
+    @PreAuthorize("isAuthenticated()")
```
Comment (blackspyek) [inline library/src/main/java/org/pollub/library/user/controller/UserController.java]: "isAuthenticated()")
instead of this
hasRole('LIBRARIAN')

Comment (blackspyek) [reply library/src/main/java/org/pollub/library/user/controller/UserController.java]: everywhere

**Review by blackspyek — CHANGES_REQUESTED** [review]
^^

File: library/src/main/java/org/pollub/library/user/controller/UserController.java (hunk 1 above)
Comment (DFraczek) [reply library/src/main/java/org/pollub/library/user/controller/UserController.java]: fixed

**Review by blackspyek — APPROVED** [review]

**Comment by blackspyek** [comment]
LGTM

**Pull request merged**